modules/hydrodyn/src/Morison.f90

```diff
@@ -2198,6 +2206,106 @@ SUBROUTINE Morison_Init( InitInp, u, p, x, xd, z, OtherState, y, m, Interval, In
    !   END SUBROUTINE
 
 END SUBROUTINE Morison_Init
+subroutine VisMeshSetup(u,p,y,m,InitOut,ErrStat,ErrMsg)
+   type(Morison_InputType),      intent(inout)  :: u
+   type(Morison_ParameterType),  intent(in   )  :: p
+   type(Morison_OutputType),     intent(inout)  :: y
+   type(Morison_MiscVarType),    intent(inout)  :: m
+   type(Morison_InitOutputType), intent(inout)  :: InitOut
+   integer(IntKi),               intent(  out)  :: ErrStat
+   character(*),                 intent(  out)  :: ErrMsg
+
+   integer(IntKi)          :: TotNodes                ! total nodes in all elements (may differ from p%NNodes due to overlaps)
+   integer(IntKi)          :: TotElems                ! total number of elements
+   integer(IntKi)          :: NdIdx, iMem, iNd, NdNum ! indexing
+   real(ReKi)              :: NdPos(3),Pos1(3),Pos2(3)
+   real(R8Ki)              :: MemberOrient(3,3)
+   real(R8Ki)              :: Theta(3)                ! Euler rotations
+   integer(IntKi)          :: ErrStat2
+   character(ErrMsgLen)    :: ErrMsg2
+   character(*), parameter :: RoutineName = 'VisMeshSetup'
+
+   ErrStat = ErrID_None
+   ErrMsg  = ""
+
+   ! Total number of nodes = sum of all member nodes
+   ! Total number of elements = sum of all member elements
+   TotNodes=0
+   TotElems=0
+   do iMem=1,size(p%Members)
+      TotElems = TotElems + p%Members(iMem)%NElements
+      TotNodes = TotNodes + size(p%Members(iMem)%NodeIndx)
+   enddo
+
+   ! Storage for the radius associated with each node
+   call AllocAry( InitOut%MorisonVisRad, TotNodes, 'MorisonVisRad', ErrStat2, ErrMsg2)
+   if (Failed())  return
+
+   call MeshCreate( BlankMesh    = y%VisMesh,         &
+                    IOS          = COMPONENT_OUTPUT,  &
+                    Nnodes       = TotNodes,          &
+                    ErrStat      = ErrStat2,          &
+                    ErrMess      = ErrMsg2,           &
+                    TranslationDisp = .TRUE.,         &
+                    Orientation     = .TRUE.          )
```

Comment:
Would it be better if this VisMesh had the same fields as Morison input point mesh (including velocities and accelerations)?

Reply:
I suspect we don't need the velocities in the VisMesh -- it's pretty rare to use them during visualization and they still exist in the Morison mesh.  Switching to adding the connectivity to the point meshes will include that information at that point.